dfget/core/core.go

```diff
@@ -115,21 +120,21 @@ func launchPeerServer(cfg *config.Config) (err error) {
 }
 
 func registerToSuperNode(cfg *config.Config, register regist.SupernodeRegister) (
```

Comment:
The `panics` in this function will be captured and ignore, the `p2p` pattern will be  degraded to `source` pattern, `dfget` will download the file from source server. But the new code returns a none-nil error to caller, this will break the downloading flow in `core.Start`.